### PR TITLE
Copy/paste tests

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: test/copyPaste.test.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        col2: 'a2'
+      },
+      {
+        col1: 3,
+        col2: 'a3'
+      }
+    ];
+  });
+
+  const columns = useMemo((): readonly Column<Row>[] => {
```
Comment (nstepien) [inline test/copyPaste.test.tsx:76]: Could define the rows and columns outside the component

File: test/copyPaste.test.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      }
+    ];
+  }, []);
+
+  function onPaste({ sourceColumnKey, sourceRow, targetColumnKey, targetRow }: PasteEvent<Row>) {
+    return { ...targetRow, [targetColumnKey]: sourceRow[sourceColumnKey as keyof Row] };
+  }
+
+  return (
+    <StrictMode>
```
Comment (nstepien) [inline test/copyPaste.test.tsx:99]: Should we render `CopyPasteTest` under `StrictMode` as well?

File: test/copyPaste.test.tsx (hunk 1 above)
Comment (amanmahajan7) [reply test/copyPaste.test.tsx]: Rows need to be updated for tests so cannot define outside

File: test/copyPaste.test.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    editor() {
+      return null;
+    }
+  }
+];
+
+const copyCellClassName = 'rdg-cell-copied';
+
+function CopyPasteTest({ allowCopyPaste = true }: { allowCopyPaste?: boolean }) {
+  const [rows, setRows] = useState((): readonly Row[] => {
```
Comment (nstepien) [inline test/copyPaste.test.tsx:28]: I mean to suggest that we can do this
```ts
const initialRows: readonly Row[] = [
  {
    col1: 1,
    col2: 'a1'
  },
  {
    col1: 2,
    col2: 'a2'
  },
  {
    col1: 3,
    col2: 'a3'
  }
];

function CopyPasteTest({ allowCopyPaste = true }: { allowCopyPaste?: boolean }) {
  const [rows, setRows] = useState(initialRows);
```

**Comment by nstepien** [comment]
Build is currently failing because of this:
https://github.com/facebook/jest/issues/11708
https://github.com/nodejs/node/pull/39593
We can downgrade node in `ci.yml` by pinning `node-version` to `16.5`, should fix it I think.

File: test/copyPaste.test.tsx (hunk 3 above)
Comment (nstepien) [reply test/copyPaste.test.tsx:28]: We can also remove the `col1` values

File: test/copyPaste.test.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+test('should not allow paste on readonly cells', () => {
+  setup();
+  userEvent.click(getCellsAtRowIndex(1)[0]);
+  copySelectedCell();
+  expect(getSelectedCell()).toHaveClass(copyCellClassName);
+  userEvent.type(document.activeElement!, '{arrowdown}');
+  pasteSelectedCell();
+  expect(getCellsAtRowIndex(2)[0]).toHaveTextContent('a3');
+});
```
Comment (nstepien) [inline test/copyPaste.test.tsx:109]: Could add a test that we allow copying a readonly cell, and pasting that value into a writable cell.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -39,7 +39,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
         with:
-          node-version: '16.x'
+          node-version: '16.5'
```
Comment (nstepien) [inline .github/workflows/ci.yml:42]: The other one needed to be changed 👀

**Review by nstepien — APPROVED** [review]

**Pull request merged**